cilium/CFP-30984-dns-proxy-ha.md

```diff
@@ -59,31 +61,32 @@ message Result {
 
 Method : UpdatesDNSRules
 
-_rpc UpdatesDNSRules(stream DNSPolicyRules) returns (Result){}_
+_rpc UpdatesDNSRules(Request) returns (stream DNSPolicyRules){}_
```

Comment:
Are we sure the flip is necessary here ?

Reply:
It seems more intuitive to me as DNSPolicyRules are being returned to the SDP.

Reply:
`UpdatesDNSRules` is intended to be invoked from SDP when there is a policy update, so cannot flip the order right ?

Reply:
Yes, `UpdateDNSRules` is intended to invoked from agent. How would the agent know about invoking the SDP ? So in this case, SDP has the responsibility to tell the cilium agent that I need the rules, instead of Cilium agent trying to find to whom it should send the rules.(if that make sense)

Reply:
We introduced tracking policy revisions per SDP instance to deal with this. On a connection, agent always sends policy information, on ACK, we track it against the connection, every policy update resulting in a revision bump we notify all SDP instances. On agent restart, we start over.